src/bindings/python/PyGroupTransform.cpp

```diff
@@ -1,164 +1,76 @@
-    TransformRcPtr childTransform = transform->getTransform(index);
-    return BuildConstPyTransform(childTransform);
-    OCIO_PYTRY_EXIT(NULL)
-}
+using TransformIterator = PyIterator<GroupTransformRcPtr, IT_TRANSFORM>;
 
-PyObject * PyOCIO_GroupTransform_getNumTransforms(PyObject * self, PyObject *)
-{
-    OCIO_PYTRY_ENTER()
-    ConstGroupTransformRcPtr transform = GetConstGroupTransform(self);
-    return PyInt_FromLong(transform->getNumTransforms());
-    OCIO_PYTRY_EXIT(NULL)
-}
+} // namespace
 
-PyObject * PyOCIO_GroupTransform_appendTransform(PyObject * self, PyObject * args)
+void bindPyGroupTransform(py::module & m)
 {
-    OCIO_PYTRY_ENTER()
-    PyObject* pytransform = 0;
-    if (!PyArg_ParseTuple(args,"O:appendTransform",
-        &pytransform)) return NULL;
-    GroupTransformRcPtr transform = GetEditableGroupTransform(self);
-    if(!IsPyTransform(pytransform))
-        throw Exception("GroupTransform.appendTransform requires a transform as the first arg.");
-    transform->appendTransform(GetEditableTransform(pytransform));
-    Py_RETURN_NONE;
-    OCIO_PYTRY_EXIT(NULL)
-}
-
+    GroupTransformRcPtr DEFAULT = GroupTransform::Create();
+
+    auto cls = py::class_<GroupTransform, 
+                          GroupTransformRcPtr /* holder */, 
+                          Transform /* base */>(m, "GroupTransform")
+        .def(py::init(&GroupTransform::Create))
+        .def(py::init([](const std::vector<TransformRcPtr> transforms, 
+                         TransformDirection dir)
+            {
+                GroupTransformRcPtr p = GroupTransform::Create();
+                for(auto & t : transforms)
+                {
+                    p->appendTransform(t);
+                }
+                p->setDirection(dir);
+                return p;
+            }), 
+             "transforms"_a = std::vector<ConstTransformRcPtr>{},
+             "dir"_a = DEFAULT->getDirection())
+
+        .def("__iter__", [](GroupTransformRcPtr & self) { return TransformIterator(self); })
+        .def("__len__", &GroupTransform::getNumTransforms)
+        .def("__getitem__", 
+             (ConstTransformRcPtr (GroupTransform::*)(int) const) &GroupTransform::getTransform, 
+             "index"_a)
+        .def("__getitem__", 
+             (TransformRcPtr & (GroupTransform::*)(int)) &GroupTransform::getTransform, 
+             "index"_a)
+             
+        .def("getFormatMetadata", 
+             (FormatMetadata & (GroupTransform::*)()) &GroupTransform::getFormatMetadata,
+             py::return_value_policy::reference_internal)
+        .def("getFormatMetadata", 
+             (const FormatMetadata & (GroupTransform::*)() const) 
+             &GroupTransform::getFormatMetadata,
+             py::return_value_policy::reference_internal)
+        .def("appendTransform", &GroupTransform::appendTransform, "transform"_a)
+        .def("prependTransform", &GroupTransform::prependTransform, "transform"_a);
+
```

Comment:
Yes, it's included in the Transform base class wrapper (via the `defStr(cls)` call, which itself is defined in `PyUtils.h`), exposed as `__str__` so in Python you can do `str(transform)` to get the same result `<<` provides in C++.

On this point I debated whether to bind this behavior to `__repr__` or `__str__`. `__repr__` initially seemed more logical to me, but when you get to `Config` it gets really out of hand since any time Python prints the object to the console (as a return value, etc.), it will then print the entire config contents. `__str__` is a bit more explicit, so I opted in the end to go that route.

Reply:
Perfect!